dynamic/binary_test.go

```diff
@@ -131,4 +133,45 @@ func TestBinaryUnknownFields(t *testing.T) {
 
 func binaryTranslationParty(t *testing.T, msg proto.Message) {
 	doTranslationParty(t, msg, proto.Marshal, proto.Unmarshal, (*Message).Marshal, (*Message).Unmarshal)
+	binaryTranslationPartyMarshalAppend(t, msg)
+}
+
+func binaryTranslationPartyMarshalAppend(t *testing.T, msg proto.Message) {
```

Comment:
I think you can just inline this into `binaryTranslationParty` above. I think you could make it a little easier on the eyes, too, by hoisting the append helper functions up and then doing this in a loop:

```go
marshalAppendSimple := func(m *Message) ([]byte, error) {
    // yadda yadda yadda
}
marshalAppendPrefix := func(m *Message) ([]byte, error) {
    // yadda yadda yadda
}

marshalMethods := []func(m *Message) ([]byte, error){
    (*Message).Marshal,
    marshalAppendSimple,
    marshalAppendPrefix,
}

for _, marshalFn := range marshalMethods {
    doTranslationParty(t, msg, proto.Marshal, proto.Unmarshal, marshalFn, (*Message).Unmarshal)
}
```

Reply:
sure, done